spec/unit/deployment/config_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'spec_helper'
+
+describe R10K::Deployment::Config do
+
+  describe "reading settings" do
+    # Try all permutations of string/symbol values as arguments to #setting and
+    # values in the config file
+    x = [:key, "key"]
+    matrix = x.product(x)
+
+    matrix.each do |(searchvalue, configvalue)|
+      it "treats #{searchvalue.inspect}:#{searchvalue.class} and #{configvalue.inspect}:#{configvalue.class} as equivalent" do
+        expect(YAML).to receive(:load_file).with('foo/bar').and_return(configvalue => 'some/cache')
+        subject = described_class.new('foo/bar')
+        expect(subject.setting(searchvalue)).to eq 'some/cache'
+      end
+    end
+  end
```

Comment:
Makes sure all permutations of string and symbol keys are equivalent. As part of https://tickets.puppetlabs.com/browse/RK-59 this should be rendered obsolete, so it's there to get test coverage right here, right now, but I would rather not have to do this at all. Kludgy tests for kludgy code.